src/main/java/com/uid2/admin/job/JobDispatcher.java

```diff
@@ -141,9 +134,47 @@ private void run(int maxRetries) {
                 }
             }
 
-            synchronized (JOB_LOCK) {
+            synchronized (jobLock) {
                 currentJob = null;
             }
         });
     }
+
+    public List<JobInfo> getJobQueueInfo() {
+        List<JobInfo> jobInfos = new ArrayList<>();
+
+        synchronized (jobLock) {
+            if (isExecutingJob()) {
+                jobInfos.add(new JobInfo(currentJob, true));
+            }
+            jobInfos.addAll(jobQueue.stream()
+                    .map(job -> new JobInfo(job, false))
+                    .collect(Collectors.toList()));
+        }
+        return jobInfos;
+    }
+
+    public JobInfo getExecutingJobInfo() {
+        synchronized (jobLock) {
+            return currentJob == null ? null : new JobInfo(currentJob, true);
+        }
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    public boolean isExecutingJob() {
+        return currentJob != null;
+    }
+
+    public boolean isStarted() {
+        return started;
+    }
+
+    private long getExecutionTime() {
+        synchronized (jobLock) {
```

Comment:
I suggest wee don't use lock here - prom will call this very frequently and will keep the lock busy.

Reply:
Sure will remove it